### DISCUSSION TITLE
Instructions for how to change zigbee2mqtt channel

### DISCUSSION THREAD

**Discussion by ChirpyTurnip** [Koenkk/zigbee2mqtt#7827] [General]
Hi,

Because I spent a couple of hours mucking around to find an answer to this rather simple problem I thought I would post it so that others googling in the future for the same answer won't just find posts with people asking questions like:

"How to change zigbee  channel on my zzh! stick?"
"Changing channel with zigbee2mqtt doesn't work"
"Log says Channel List: configured=15, adapter=11 - how to change channel?"

The answer is simple - it's a TWO stage process:

Step 1 - Set the channel you want 

Change your configuration.yaml to specify the channel you want;

OR

Change it in the zigbee2mqtt GUI by going to Settings > Advanced > ZigBee channel

Step 2 - Apply the change

- Stop zigbee2mqtt (either by stopping the service or stopping the docker container)
- Delete the /app/data/coordinator_backup.json file that the system uses to set the channel (this will reset your whole zigbee network)
- Restart the zigbee2mqtt service or container

It's possible that editing the coordinator_backup.json file will also work, but given that you'll need to get everything to repair again it's largely a moot point I suspect. 

It's incredible that Step 2 is so easy yet so hard to find....hopefully this post will help the next schmuck looking for answers. 

CP.

**Comment by c0mplex1** [comment]
Hi,

Thanks for the clear instructions, but after the change you still have to repair all the devices on your network, right?

**Comment by ChirpyTurnip** [comment]
Yes....I only had two sensors but having deleted them from the backup they
joined straight away when the new channel came up (I set it to allow join
as part of my config change). Luckily this was an easy job.... Soon I'll be
deploying another thirty devices and that will make it harder...

But, on the plus side, much more stable now that my two mqtt instances are
on different channels. :-)

**Comment by rocket59** [comment]
A most appreciative schmuck here! Thanks @ChirpyTurnip!

For RPI users, the coordinator_backup.json file is located at  /opt/zigbee2mqtt/data (same place as the configuration.yaml file). I simply renamed mine in case I wanted to go back to the original channel.

@Koenkk can I suggest these extra steps be included in the main site (at least for CC2531 users - don't know if the above procedure is required for other zigbee sticks, although I rather suspect so).

**Comment by KoalaWerewolf** [comment]
An addition based on having changed Zigbee-channel recently. I changed the channel from 25 since the documentation for some of my devices state that they scan 11-24.

I edited the channel in configuration.yaml and coordinator_backup.json. I did this because I expected the id's I've set to be preserved after repairing. This worked as expected, empty device list after restart, when repairing a device it joined with the id I had set previously. This meant that everything in HA just worked after a device joined again.

An interesting aspect of this process was that a few devices found the new network by themselves, without me having to re-pair them: Hue Motion Sensor, Hue Motion Outdoor Sensor, Wiser wireless switch, Frient/Develco Smart plug and IKEA Fyrtur. I suspect that Permit Join was active when they found their way back, but I didn't notice when each of these devices joined since I was pairing other devices.

**Comment by saxon-viking** [comment]
ChirpyTurnip,
Many thanks, for the info. I have changed from 11, to 25 (which is terrible in my house), now on 20 after using your tips.
Works brilliantly, I stop z2m, change channel, save, delete the coordinator backup, then start z2m. Flawless!

**Comment by ChirpyTurnip** [comment]
You're welcome! Glad it helped....I wrote it up because a good guide is always a lifesaver when you're stuck!  :-)

**Comment by djcrawleravp** [comment]
So...

Stopping the container then changing the channel in configuration.yaml and enabling "permit Join"

Then changing the channel in coordinator_backup.json

Staring the container

That will re-pair some devices?

**Reply by KE55ARD** [reply]
FWIW I just did pretty much this and most devices have rejoined by themselves!

Exact steps below:
- Changed channel in Zigbee2MQTT settings in the UI
- Stopped Zigbee2MQTT
- Copied coordinator_backup.json (with another filename just in case I want to revert)
- Edited the channel and channel_mask in coordinator_backup.json to the new channel
- Started Zigbee2MQTT

**Reply by marekpytel** [reply]
worked like a charm on SLZB-06 dongle, thank you :)

**Comment by slavo74** [comment]
How to delete coordinator_backup.json file when Linux is running on RPi? How do you edit the files in the root directory of Homebridge?

**Reply by slavo74** [reply]
I found that for Homebridge in Rpi 5 there is a coordinator_backup.json file in the /home/pi/data directory
